fhir-server/src/test/java/au/csiro/pathling/fhirpath/literal/CodingLiteralPathTest.java

```diff
@@ -0,0 +1,135 @@
+        expression,
+        inputContext);
+    final Coding literalValue = codingLiteralPath.getLiteralValue();
+    assertEquals("http://snomed.info/sct", literalValue.getSystem());
+    assertEquals("http://snomed.info/sct/32506021000036107/version/20201231",
+        literalValue.getVersion());
+    assertEquals("166056000", literalValue.getCode());
+
+    final String actualExpression = codingLiteralPath.getExpression();
+    assertEquals(expression, actualExpression);
+  }
+
+  @Test
+  void roundTripNoVersion() {
+    final String expression = "http://snomed.info/sct|166056000";
+    final CodingLiteralPath codingLiteralPath = CodingLiteralPath
+        .fromString(expression, inputContext);
+    final Coding literalValue = codingLiteralPath.getLiteralValue();
+    assertEquals("http://snomed.info/sct", literalValue.getSystem());
+    assertNull(literalValue.getVersion());
+    assertEquals("166056000", literalValue.getCode());
+
+    final String actualExpression = codingLiteralPath.getExpression();
+    assertEquals(expression, actualExpression);
+  }
+
+  @Test
+  void fromStringWithQuotedComponent() {
+    final String expression =
+        "http://snomed.info/sct|http://snomed.info/sct/32506021000036107/version/20201231|"
+            + "'397956004 |Prosthetic arthroplasty of the hip|: 363704007 |Procedure site| = "
+            + "( 24136001 |Hip joint structure|: 272741003 |Laterality| =  7771000 |Left| )'";
+    final CodingLiteralPath codingLiteralPath = CodingLiteralPath
+        .fromString(expression, inputContext);
+    final Coding literalValue = codingLiteralPath.getLiteralValue();
+    assertEquals("http://snomed.info/sct", literalValue.getSystem());
+    assertEquals("http://snomed.info/sct/32506021000036107/version/20201231",
+        literalValue.getVersion());
+    assertEquals(
+        "397956004 |Prosthetic arthroplasty of the hip|: 363704007 |Procedure site| = "
+            + "( 24136001 |Hip joint structure|: 272741003 |Laterality| =  7771000 |Left| )",
+        literalValue.getCode());
+
+    final String actualExpression = codingLiteralPath.getExpression();
+    assertEquals(expression, actualExpression);
+  }
+
+  @Test
+  void fromStringWithQuotedComponentWithComma() {
+    final String expression =
+        "http://snomed.info/sct|http://snomed.info/sct/32506021000036107/version/20201231|'46,2'";
+    final CodingLiteralPath codingLiteralPath = CodingLiteralPath
+        .fromString(expression, inputContext);
+    final Coding literalValue = codingLiteralPath.getLiteralValue();
+    assertEquals("http://snomed.info/sct", literalValue.getSystem());
+    assertEquals("http://snomed.info/sct/32506021000036107/version/20201231",
+        literalValue.getVersion());
+    assertEquals("46,2", literalValue.getCode());
+
+    final String actualExpression = codingLiteralPath.getExpression();
+    assertEquals(expression, actualExpression);
+  }
```

Comment:
But is this correct? 
If code value "46,2" should be quoted and escaped, should not he actualExpression have be:
" ... |`46\\,2"? 
Also to test escaping should the expression be " ... |`46\\,2" as well?

Reply:
I don't think commas need to be escaped in the context of a string literal: http://hl7.org/fhirpath/#string. They are, however, problematic _outside_ of a string literal.

If we had a Coding with a code that contained a comma that we wanted to encode as a Coding literal, we would need to quote the code to ensure that the parser didn't interpret it as the end of the Coding literal (and maybe the start of a new argument, if it was in the context of a function's arguments). The same goes for whitespace, parentheses and the pipe operator.

So `NEEDS_QUOTING` needs to be the minimal set of characters that indicate that the component will need to be quoted so that it doesn't fall foul of the patterns in `FhirPath.g4`.

Reply:
Ok, I guess I did not fully understand what needs to be quoted/escapes.
But do we have a test case for such a coding the needs escaping?

Reply:
I have added more tests in 8ea952a.